numpy/lib/function_base.py

```diff
@@ -2818,7 +2821,12 @@ def corrcoef(x, y=None, rowvar=True, bias=np._NoValue, ddof=np._NoValue, *,
         # 2015-03-15, 1.10
         warnings.warn('bias and ddof have no effect and are deprecated',
                       DeprecationWarning, stacklevel=3)
-    c = cov(x, y, rowvar, dtype=dtype)
+    if x is None and cov is None:
+        raise ValueError("'x' or 'cov' must be passed as a parameter")
```

Comment:
This sould be a `TypeError`, it should be similar to what you currently get for `np.corrcoef()` after all.

Reply:
I don't think you check for _both_ being passed.  We need to check that exactly one of them is passed only.